source/ceylon/http/server/internal/RequestImpl.ceylon

```diff
@@ -67,14 +73,43 @@ class RequestImpl(HttpServerExchange exchange,
     contentType => getHeader(Headers.contentType.string);
 
     header(String name) => getHeader(name);
-    
+
+    [Locale+] getLocales() {
+        value acceptLanguage = exchange.requestHeaders.get(Headers.acceptLanguage.string);
+        if (!exists acceptLanguage) {
+            return [systemLocale];
+        }
+        if (acceptLanguage.empty) {
+            return [systemLocale];
+        }
+        value parsedResults = QValueParser.parse(acceptLanguage);
+        ArrayList<Locale> locales = ArrayList<Locale>();
+        for (qvalueResult in parsedResults) {
+            for (res in qvalueResult) {
+                if (is Locale nextLocale = createLocale(res.\ivalue.string)) {
+                    locales.add(nextLocale);
+                }
+            }
+        }
+        if (locales.empty) {
```

Comment:
Maybe flip the if statement around and use `if (nonempty locales) {` instead, then you can get rid of the assert below.